app/submitter/encrypter.py

```diff
@@ -0,0 +1,74 @@
+from cryptography.hazmat.backends.openssl.backend import backend
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+from app import settings
+
+import jwt
+import os
+import base64
+
+
+class Encrypter (object):
+    def __init__(self):
+        private_key_bytes = self._to_bytes(settings.EQ_SR_PRIVATE_ENCRYPTION_KEY)
+        public_key_bytes = self._to_bytes(settings.EQ_SDX_PUBLIC_KEY)
+        self.private_key = serialization.load_pem_private_key(private_key_bytes,
+                                                              password=self._to_bytes(settings.EQ_SR_PRIVATE_ENCRYPTION_KEY_PASSWORD),
+                                                              backend=backend)
+
+        self.public_key = serialization.load_pem_public_key(public_key_bytes, backend=backend)
+
+        # first generate a random key
+        self.cek = os.urandom(32)  # 256 bit random CEK
+
+        # now generate a random IV
+        self.iv = os.urandom(12)  # 96 bit random IV
+
+    def _to_bytes(self, bytes_or_str):
+        if isinstance(bytes_or_str, str):
+            value = bytes_or_str.encode()
+        else:
+            value = bytes_or_str
+        return value
+
+    def _jwe_protected_header(self):
+        return self._base_64_encode(b'{"alg":"RSA-OAEP","enc":"A256GCM"}')
+
+    def _encrypted_key(self, cek):
+        ciphertext = self.public_key.encrypt(cek, padding.OAEP(mgf=padding.MGF1(algorithm=hashes.SHA1()), algorithm=hashes.SHA1(), label=None))
+        return self._base_64_encode(ciphertext)
+
+    def _encode_iv(self, iv):
+        return self._base_64_encode(iv)
+
+    def _base_64_encode(self, text):
+        # strip the trailing = as they are padding to make the result a multiple of 4
+        # the RFC does the same, as do other base64 libraries so this is a safe operation
+        return base64.urlsafe_b64encode(text).decode().strip("=").encode()
+
+    def _encode_and_signed(self, payload):
+        return jwt.encode(payload, self.private_key, algorithm="RS256", headers={'kid': 'EDCRRM', 'typ': 'jwt'})
```

Comment:
Need to agree a different KID for this, not EDCRRM, and make it a const for the class? Perhaps EDCSR?